src/api/base/index.ts

```diff
@@ -144,4 +114,41 @@ export default class BaseApi {
       body: data !== undefined ? JSON.stringify(data) : data,
     })
   }
+
+  private async fetchMultipleEnvsRecursive(
+    url: string,
+    method: 'GET' | 'POST' | 'DELETE',
+    fetchFn: BaseApi['fetch'],
+    getApiUrl: BaseApi['API_URL_GETTER'],
+    options: Omit<Options, 'env' | 'chainId'> = {},
+    envs: Env[],
+    chainId: SupportedChainId,
+    data?: unknown
```

Comment:
any reason why you went for so many arguments, instead of using a single options param?

Reply:
Because I started from the initial call.
Thought about changing to obj params but forgot.
Will update.